### PR TITLE
Prefetch default included resources

### PR DESCRIPTION
**Pull request by SafaAlfulaij**
## Description of the Change
Pass the serializer to `select_related` and `prefetch_related` default `included_resources`. Without this, only resources specified by `?include` are used.

## Checklist

- [x] PR only contains one change (considered splitting up PR)
- [x] unit-test added
- [ ] documentation updated
- [x] `CHANGELOG.md` updated (only for user relevant changes)
- [x] author name in `AUTHORS`

### CODE REVIEW

**Comment by SafaAlfulaij** [comment]
Sorry, but I have no idea why the test is failing.
I've never done tests.

**Comment by sliverc** [comment]
Thanks for your contribution. PR number 900... wow! This is certainly a good addition to add support for default included resources. We might even consider this to be a bug that this has not been the case in the first place...

In terms of tests for quality assurance all PRs in DJA need to have a test to prove that the change actually does what it is supposed too. 

It is never too late to start learning about testing though... for instance this is a [good tutorial](https://realpython.com/python-testing/) but there are many others out there.

It also seems with your change you discovered a potential another problem that inlcudes currently do not work when using relationship view. See [failing test](https://github.com/django-json-api/django-rest-framework-json-api/pull/900/checks?check_run_id=2211276448) where a relationship view now prefetches default resources and fails.

Some more research needs to be done. If you think this is a bit too hard of a task we can also convert this PR to a issue report and somewhere else can have a look at it and have your PR as a reference.

Do you wanna give it or shot or shall we convert it to a bug report? What do you think?

File: rest_framework_json_api/serializers.py [hunk 1]
```diff
@@ -137,7 +137,7 @@ def validate_path(serializer_class, field_path, path):
                 validate_path(this_included_serializer, new_included_field_path, path)
 
         if request and view:
-            included_resources = get_included_resources(request)
+            included_resources = get_included_resources(request, self)
             for included_field_name in included_resources:
                 included_field_path = included_field_name.split(".")
                 this_serializer_class = view.get_serializer_class()
```
[the remark this answers is not in the document]
Comment (sliverc) [reply rest_framework_json_api/serializers.py:143]: This is a regression from PR #860 and the error was in the code base before but was not discovered as no test was testing includes on relationship views.


A bit further down on line 143 a snippet like this should solve the issue (needs to be tested):

```python
                if "related_field" in view.kwargs:
                    this_serializer_class = view.get_related_serializer_class()
                else:
                    this_serializer_class = view.get_serializer_class()
```

**Comment by SafaAlfulaij** [comment]
I'm not sure if the test environment is different than the normal environment, but `/entries?fields[entry]=comments&page[size]=25` executes exactly 3 queries. I don't know why the test environment ignores `fields[entry]=comments`.

Anyway, this is the best I can reach in testing. If anyone is willing to do more, please feel free to take over.

**Comment by SafaAlfulaij** [comment]
So the tests use the plural (normal I believe) form of resources, while the end-user example uses the singular form.
Should be fixed now.

File: rest_framework_json_api/serializers.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (sliverc) [reply rest_framework_json_api/serializers.py]: As this change is not directly related with this PR I extracted it into another PR #914 Once this is merged I will be able to take another look at this PR.

**Comment by sliverc** [comment]
@SafaAlfulaij As of changes merged in #914 could you rebase this to master?

**Comment by simkimsia** [comment]
It seems that this is the only issue preventing #911. I can free up sometime this coming Wednesday to look at rebasing this to master if that helps?

**Comment by sliverc** [comment]
@simkimsia Thanks for your offer. Really appreciated. I got around today to look at this PR and hope a new version of DJA will be out soon.

**Review by sliverc — APPROVED** [review]
@SafaAlfulaij I made some small adjustment and the PR is now ready for merging. Thanks for your work!

**Pull request merged**